Add non-root user to the docker group for non-root Docker access

polmoorx opened the pull request:
#### Merge Checklist  <!-- REQUIRED -->
**All** boxes should be checked before merging the PR
- [x] The changes in the PR have been built and tested
- [] cgmanifest file has been updated if required
- [x] Ready to merge

#### Description <!-- REQUIRED -->
<!-- Please include a summary of the changes and the related issue. List any dependencies that are required for this change. -->
The non-root user couldn’t run Docker commands without sudo because it was not part of the 'docker' group. Added a new script to automatically detect the default non-root user and add it to the docker group, This allows non-root to use Docker without sudo while keeping existing sudo behavior intact.

This allows guest to use Docker without sudo while keeping existing sudo behavior intact.
<!-- Fixes # (issue) -->

#### Any Newly Introduced Dependencies
<!-- Please describe any newly introduced 3rd party dependencies in this change. List their name, license information and how they are used in the project. -->
NO
#### How Has This Been Tested? <!-- REQUIRED -->
<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->
Manually tested

andy-vm reviewed (changes requested):
this file is add-sudoer.sh, should not be mixed up with docker group, please add another script for docker group purpose

polmoorx commented:
> this file is add-sudoer.sh, should not be mixed up with docker group, please add another script for docker group purpose

Hi @andy-vm,

Thanks for the review. I’ve moved the Docker group logic out of add-sudoer.sh into a new file called add-docker-group.sh. The new script handles adding the user to the Docker group. Image JSONs have been updated to include the script.

andy-vm commented:
does the new script have execution permissions added? please build one new image and test 
```
chmod +x script
```

polmoorx commented:
> does the new script have execution permissions added? please build one new image and test
> 
> ```
> chmod +x script
> ```

yes @andy-vm it has  execution permissions and tested locally, working as expected.

andy-vm commented on toolkit/imageconfigs/additionalconfigs/add-docker-group.sh:
```diff
@@ -0,0 +1,10 @@
+set -e
+
+# Detect the default non-root user
+user=$(awk -F: '$3 >= 1000 && $1 != "nobody" {print $1; exit}' /etc/passwd)
```
this script returns first non root user, not guest
we have guest user in add-sudoer.sh, we can just use guest here

polmoorx replied:
Hi @andy-vm,
Updated the script to use the hardcoded guest user instead of detecting the first non-root user, to keep it consistent with add-sudoer.sh.

andy-vm reviewed (dismissed):
LGTM

cheeyanglee commented:
why is this needed for most of the image? only -dev image have default guest user

polmoorx commented:
> why is this needed for most of the image? only -dev image have default guest user

Thanks for the review. Updated the PR, new script now automatically detects the default non-root user (e.g., guest in dev images, others in non-RT images) and adds that user to the docker group. This addresses the issue consistently across all image types.

cheeyanglee commented:
> > why is this needed for most of the image? only -dev image have default guest user
> 
> Thanks for the review. Updated the PR, new script now automatically detects the default non-root user (e.g., guest in dev images, others in non-RT images) and adds that user to the docker group. This addresses the issue consistently across all image types.

default user doesnt mean it need docker group access.  what is the problem this PR try to solve?

polmoorx commented:
> > > why is this needed for most of the image? only -dev image have default guest user
> > 
> > 
> > Thanks for the review. Updated the PR, new script now automatically detects the default non-root user (e.g., guest in dev images, others in non-RT images) and adds that user to the docker group. This addresses the issue consistently across all image types.
> 
> default user doesnt mean it need docker group access. what is the problem this PR try to solve?

Based on ticket **ITEP-81113**, the normal user in the images is unable to run Docker commands because it is not part of the docker group. As a result, Docker operations fail with a “permission denied” error.

This PR resolves the issue by adding the primary non-root user to the docker group, ensuring Docker works as expected for the intended user.

cheeyanglee commented:
> his PR resolves the issue by adding the primary non-root user to the docker group, ensuring Docker works as expected for the intended user.

this is fine for -dev image with default non-root user (guest) added [1] as we added the guest user for development and test purpose only. 
This is not appropriate for image without default non-root user, these permissions should be manage by cloud-init or the root user.  

[1] https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/573/files#diff-5470aff6ec3ad901ddaf3e5a5c221137fb3dc629a924df7ad94c306a321fc65aR132

polmoorx committed:
Add guest user to the docker group for docker access

The guest user couldn’t run Docker commands without sudo because it
was not part of the 'docker' group. A new script added to ensure the
guest user added to the docker group, This allows guest user to use
Docker without sudo while keeping existing sudo behavior intact.

Signed-off-by: Polmoorx Shiva Kumar <polmoorx.shiva.kumar@intel.com>

polmoorx commented:
> > his PR resolves the issue by adding the primary non-root user to the docker group, ensuring Docker works as expected for the intended user.
> 
> this is fine for -dev image with default non-root user (guest) added [1] as we added the guest user for development and test purpose only. This is not appropriate for image without default non-root user, these permissions should be manage by cloud-init or the root user.
> 
> [1] https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/573/files#diff-5470aff6ec3ad901ddaf3e5a5c221137fb3dc629a924df7ad94c306a321fc65aR132

Hi @cheeyanglee,

As per the latest comment in ITEP-81113 the issue is not ibserving in cloud-init and the issue is only observed in dev image and in dev image the the gusest user  is hardcoded hence the PR is fixed for dev images alone and working as expected

> > his PR resolves the issue by adding the primary non-root user to the docker group, ensuring Docker works as expected for the intended user.
> 
> this is fine for -dev image with default non-root user (guest) added [1] as we added the guest user for development and test purpose only. This is not appropriate for image without default non-root user, these permissions should be manage by cloud-init or the root user.
> 
> [1] https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/573/files#diff-5470aff6ec3ad901ddaf3e5a5c221137fb3dc629a924df7ad94c306a321fc65aR132

Hi @cheeyanglee,
Thank you.
Based on the latest update in **ITEP-81113**, the issue isn’t seen in cloud-init. It only appears in the dev images, where the guest user is added for development and testing.
Hence, the PR has been updated to fix the dev images by adding the guest user to the docker group.
**For reference please see below attached snippet from dev-image.**
<img width="532" height="249" alt="image" src="https://github.com/user-attachments/assets/1e241004-096a-4f8f-be53-304b4cffc6d7" />
